src/integrator.jl

```diff
@@ -46,6 +46,11 @@ function step(lf::Leapfrog{F},
             θ, r = θ_new, r_new
             n_valid = n_valid + 1
         else
+            # Reverse half leapfrog step for r when breaking
+            #  the loop immaturely.
+            if i > 1 && i < _n_steps
```

Comment:
Should it be `i <= _n_steps` here?

Reply:
seems that `i < _n_steps` is sufficient because of 

https://github.com/TuringLang/AdvancedHMC.jl/blob/10659b225f5ddc8c4614cdd62cc4bc27799c10a8/src/integrator.jl#L44

Reply:
Oh yes! Sorry I forget this condition above. Then do we need `i > 1` here?

Reply:
yes; because if we break out of the loop when `i = 1`, `θ` and  `r` are not updated at all, see

https://github.com/TuringLang/AdvancedHMC.jl/blob/10659b225f5ddc8c4614cdd62cc4bc27799c10a8/src/integrator.jl#L46

Reply:
Great! It's clear now!